packages/compile-solidity/src/run.ts

```diff
@@ -412,55 +480,58 @@ function processContracts({
             sourcePath: transformedSourcePath,
             contents: source
           }
-        }) => ({
-          contractName,
-          abi: orderABI({ abi, contractName, ast }),
-          metadata,
-          devdoc,
-          userdoc,
-          sourcePath: originalSourcePaths[transformedSourcePath],
-          source,
-          sourceMap,
-          deployedSourceMap: (deployedBytecodeInfo || {}).sourceMap,
-          ast,
-          legacyAST,
-          bytecode: zeroLinkReferences({
-            bytes: bytecode,
-            linkReferences: formatLinkReferences(linkReferences)
-          }),
-          deployedBytecode: zeroLinkReferences({
-            bytes: (deployedBytecodeInfo || {}).object,
-            linkReferences: formatLinkReferences(
-              (deployedBytecodeInfo || {}).linkReferences
-            )
-          }),
-          immutableReferences: (deployedBytecodeInfo || {}).immutableReferences,
-          //ideally immutable references would be part of the deployedBytecode object,
-          //but compatibility makes that impossible
-          generatedSources,
-          deployedGeneratedSources: (deployedBytecodeInfo || {})
-            .generatedSources,
-          compiler: {
-            name: "solc",
-            version: solcVersion
-          }
-        })
+        }) => {
+          return {
+            contractName,
+            abi: orderABI({ abi, contractName, ast }),
+            metadata,
+            devdoc,
+            userdoc,
+            sourcePath: originalSourcePaths[transformedSourcePath],
+            source,
+            sourceMap,
+            deployedSourceMap: (deployedBytecodeInfo || {}).sourceMap,
+            ast,
+            legacyAST,
+            bytecode: zeroLinkReferences({
+              bytes: bytecode,
+              linkReferences: formatLinkReferences(linkReferences)
+            }),
+            deployedBytecode: zeroLinkReferences({
+              bytes: (deployedBytecodeInfo || {}).object,
+              linkReferences: formatLinkReferences(
+                (deployedBytecodeInfo || {}).linkReferences
+              )
+            }),
+            immutableReferences: (deployedBytecodeInfo || {})
+              .immutableReferences,
```

Comment:
this would be a good place to use the coalescing operator
```suggestion
            immutableReferences: deployedBytecodeInfo?.immutableReferences,
```